python/dgl/dataloading/pytorch/__init__.py

```diff
@@ -42,16 +45,44 @@ class _ScalarDataBatcher(th.utils.data.IterableDataset):
     is passed in.
     """
     def __init__(self, dataset, shuffle=False, batch_size=1,
-                 drop_last=False):
+                 drop_last=False, use_ddp=False, ddp_seed=0):
         super(_ScalarDataBatcher).__init__()
         self.dataset = dataset
         self.batch_size = batch_size
         self.shuffle = shuffle
         self.drop_last = drop_last
+        self.use_ddp = use_ddp
+        if use_ddp:
+            self.rank = dist.get_rank()
+            self.num_replicas = dist.get_world_size()
+            self.seed = ddp_seed
+            self.epoch = 0
+            # The following code (and the idea of cross-process shuffling with the same seed)
+            # comes from PyTorch.  See torch/utils/data/distributed.py for details.
+
+            # If the dataset length is evenly divisible by # of replicas, then there
+            # is no need to drop any data, since the dataset will be split equally.
```

Comment:
"data" -> "sample"

Reply:
I copied the comments from PyTorch haha.